src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,264 @@
+    """
+    df_ploidy = pd.read_csv(filename, sep="\t")
+    df_ploidy.loc[df_ploidy["chrX_CopyNumber"] <= 2, "chrX_CopyNumber_rounded"] = 1
+    df_ploidy.loc[df_ploidy["chrX_CopyNumber"] > 2, "chrX_CopyNumber_rounded"] = 2
+    return df_ploidy
+
+
+def read_bincov_median(filename: str) -> pd.DataFrame:
+    """
+    Median coverage (bincov_median)
+    Args:
+        filename: bincov_median output from EvidenceQC (medianCov.transposed) in a tab separated format
+
+    Returns:
+        a pandas DataFrame containing the transposed table given in the input file.
+    """
+    df_median = pd.read_csv(filename, sep="\t").T
+    df_median = df_median.rename_axis(ID_COL).reset_index()
+    df_median.columns = [ID_COL, "median_coverage"]
+    df_median = df_median.reset_index(drop=True)
+    return df_median
+
+
+def read_wgd_scores(filename: str) -> pd.DataFrame:
+    return pd.read_csv(filename, sep="\t")
+
+
+def read_non_diploid(filename: str) -> pd.DataFrame:
+    """
+    EvidenceQC → untar → /ploidy_est/binwise_CNV_qValues.bed.gz →
+    count number of bins with q-value < 0.05 for each sample
+
+    Args:
+        filename: a tab-delimited file containing binwise CNV qValues.
+
+    Returns:
+        A pandas DataFrame containing the number of bins with q-value < 0.05 for each sample.
+    """
+    df_non_diploid = pd.read_csv(filename, sep="\t").T
+    df_non_diploid = df_non_diploid.T.iloc[0:, 3:]  # .rename_axis("#ID").reset_index()
+    new_c = df_non_diploid.iloc[0:, 0:]
+    nondiploid_counts = new_c[new_c < 0.05].count()
+    nondiploid_counts_df = pd.DataFrame(nondiploid_counts, columns=["nondiploid_counts"]).rename_axis(
+        "#ID").reset_index()
+    return nondiploid_counts_df
+
+
+def read_melt_insert_size(filename: str) -> array:
+    """
+    Reads MELT insert size into an array.
+    Args:
+        filename: melt_insert_size in sample data table
+
+    Returns:
+        Array[Float] of melt_insert_size for each sample
+    """
+    insert_size = pd.read_csv(filename, sep="\t")
+    melt_insert_size_array = pd.DataFrame(insert_size, columns=["melt_insert_size"]).astype(float)
+    return melt_insert_size_array
+
+
+def get_col_name(caller: str, outlier_type: str) -> str:
+    return f"{caller}_{outlier_type}_outlier"
+
+
+def read_outlier(filename: str, outlier_col_label: str, idx=10) -> pd.DataFrame:
+    """
+
+    Args:
+        filename: A tab-delimited file containing the output of
+        Evidence QC on a caller (e.g., Manta) with low/high.
+
+        outlier_col_label: Column header for the outliers.
+
+        idx: The index of the cell in the array containing sample
+        URI split by `/` that contains sample ID.
+    Returns:
+        A pandas DataFrame containing the number of times a sample
+        appears in the QC of a caller output.
+    """
+    df = pd.read_csv(filename, sep="\t")
+    if df["Outlier_Sample"].empty:
+        df[ID_COL] = df.apply(lambda _: "NA", axis=0)
```

Comment:
You had set the ID to empty string, I have seen issues with that in particular when a line is trimmed before split by a delimiter. So, I changed the empty string with `NA`; however, I'm not sure what is the best approach here. Do you need this row?

Reply:
@VJalili I think we need this row so that it can return `'0'` for the empty files (the files that we did not have any types of outliers).

Reply:
Thanks! That makes sense to me.

Reply:
@epiercehoffman It is therefore necessary to check if the file is blank. Empty files should also include outlier columns. Even empty files need IDs when merging based on the `'#ID'` key at the end.  I think it will be more clear once you run the code using the empty and non-empty files.

Reply:
Do you end up with an extra row for `NA` in the case of an empty file due to the outer join later? I see an extra row with no ID in your example output, which we would want to remove